Mathlib/Combinatorics/Additive/RuzsaCovering.lean

```diff
@@ -4,68 +4,72 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yaël Dillies
 -/
 import Mathlib.Algebra.Group.Pointwise.Finset.Basic
+import Mathlib.Data.Real.Basic
 import Mathlib.SetTheory.Cardinal.Finite
+import Mathlib.Tactic.Positivity.Finset
 
 /-!
 # Ruzsa's covering lemma
 
-This file proves the Ruzsa covering lemma. This says that, for `s`, `t` finsets, we can cover `s`
-with at most `(s + t).card / t.card` copies of `t - t`.
-
-## TODO
-
-Merge this file with other prerequisites to Freiman's theorem once we have them.
+This file proves the Ruzsa covering lemma. This says that, for `A`, `B` finsets, we can cover `A`
+with at most `#(A + B) / #B` copies of `B - B`.
 -/
 
+open scoped Pointwise
 
-open Pointwise
+variable {G : Type*} [Group G] {K : ℝ}
 
 namespace Finset
-
-variable {α : Type*} [DecidableEq α] [CommGroup α] (s : Finset α) {t : Finset α}
+variable [DecidableEq G] {A B : Finset G}
 
 /-- **Ruzsa's covering lemma**. -/
 @[to_additive "**Ruzsa's covering lemma**"]
-theorem exists_subset_mul_div (ht : t.Nonempty) :
-    ∃ u : Finset α, u.card * t.card ≤ (s * t).card ∧ s ⊆ u * t / t := by
-  haveI : ∀ u, Decidable ((u : Set α).PairwiseDisjoint (· • t)) := fun u ↦ Classical.dec _
-  set C := s.powerset.filter fun u ↦ u.toSet.PairwiseDisjoint (· • t)
-  obtain ⟨u, hu, hCmax⟩ := C.exists_maximal (filter_nonempty_iff.2
-    ⟨∅, empty_mem_powerset _, by rw [coe_empty]; exact Set.pairwiseDisjoint_empty⟩)
-  rw [mem_filter, mem_powerset] at hu
-  refine ⟨u,
-    (card_mul_iff.2 <| pairwiseDisjoint_smul_iff.1 hu.2).ge.trans
-      (card_le_card <| mul_subset_mul_right hu.1),
-    fun a ha ↦ ?_⟩
+theorem ruzsa_covering_mul (hB : B.Nonempty) (hK : #(A * B) ≤ K * #B) :
+    ∃ F ⊆ A, #F ≤ K ∧ A ⊆ F * B / B := by
```

Comment:
Would be nice to state it as:
```suggestion
theorem ruzsa_covering_mul (hB : B.Nonempty) (hK : #(A * B) ≤ K * #B) :
    ∃ F ⊆ A, #F ≤ K ∧ A ⊆ F * (B / B) := by
```
so that it matches the docstring more obviously